src/psyclone/generator.py

```diff
@@ -98,29 +98,35 @@ def handle_script(script_name, info, function_name, is_optional=False):
 
     '''
     # pylint: disable=too-many-locals
-    sys_path_appended = False
+    sys_path_prepended = False
     try:
-        # a script file has been provided, check it exists
-        if not os.path.isfile(script_name) and not any(
-               os.path.isfile(os.path.join(p, script_name)) for p in sys.path):
-            raise GenerationError(
-                f"generator: script file '{script_name}' not found")
         filepath, filename = os.path.split(script_name)
-        if filepath:
-            # a path has been provided, so add it to the python search path
-            sys_path_appended = True
-            sys.path.append(filepath)
-        filename, fileext = os.path.splitext(filename)
+        module_name, fileext = os.path.splitext(filename)
+        # the file must either be:
+        # a) at the given path or, given no path, in the current directory; or
+        # b) given no path, in the system path
+        if not (os.path.isfile(script_name) or
+                not filepath and any (os.path.isfile(os.path.join(p, filename))
+                                      for p in sys.path)):
+                raise GenerationError(
+                    f"generator: script file '{script_name}' not found")
+        # the file must have the .py extension
         if fileext != '.py':
             raise GenerationError(
                 f"generator: expected the script file '{filename}' to have "
                 f"the '.py' extension")
+        # prepend file path - if none, the empty string equates to the current
+        # working directory - to the system path to guarantee we find the user
+        # provided module instead of a similarly named module that might
+        # already exist elsewhere in the system path
+        sys_path_prepended = True
+        sys.path.insert(0, filepath)
         try:
-            transmod = __import__(filename)
+            transmod = __import__(module_name)
         except Exception as error:
             raise GenerationError(
-                f"generator: attempted to import '{filename}' but a problem "
-                f"was found: {error}") from error
+                f"generator: attempted to import '{module_name}' but a "
```

Comment:
Not yours but looking at the tests, I think a bit more context would be useful in this message, e.g.: "...attempted to import specified PSyclone optimisation script '{module_name}' but...." and again for the error messages below.

Reply:
Done.